File: java/resistor-color/src/main/java/ResistorColor.java
import java.util.List;

class ResistorColor {
    private static final List<String> colors = List.of(
            "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white");

    int colorCode(String color) {
        return colors.indexOf(color);
    }

    String[] colors() {
        return colors.toArray(String[]::new);
    }
}

File: c/resistor-color/src/resistor_color.c
#include "resistor_color.h"

int color_code(resistor_band_t color) {
    return (int) color;
}


const resistor_band_t* colors() {
    static resistor_band_t result[] = {
            BLACK, BROWN, RED, ORANGE, YELLOW,
            GREEN, BLUE, VIOLET, GREY, WHITE
    };

    return result;
}